### ISSUE TITLE
bug with java methods and primitive types

### ISSUE THREAD

**Issue by JonathanStarup** [flix/flix#8796]
It looks like method lookup thinks that we do automatic coercion
```
import java.lang.Math
def main(): Float32 \ IO = Math.max(32f32, 0)
```
```
Verifier error:
Mismatched types near ..\test.flix:2:28: tpe1 = Int32, class = float
```

**Comment by magnus-madsen** [comment]
Yeah. We need to support it or emit an error.

**Comment by JonathanStarup** [comment]
Fixed in #6961
